src/entities/removeLiquidity/index.ts

```diff
@@ -23,6 +25,13 @@ export class RemoveLiquidity implements RemoveLiquidityBase {
     ): Promise<RemoveLiquidityQueryOutput> {
         this.inputValidator.validateRemoveLiquidity(input, poolState);
         switch (poolState.vaultVersion) {
+            case 0: {
+                const removeLiquidity = new RemoveLiquidityCowAmm();
```

Comment:
Similar to add - wonder if we should check for version?

Reply:
Similar response - pool version shouldn't affect the code as long as interfaces keep the same, because proportional adds/removes are not dependent on pool model.